### PR TITLE
Add command line interface for controlling OpenAPI spec version used in codegen

### PR DESCRIPTION
**Pull request by mzeitlin11**
This PR adds a small command line interface for the codegen process. The main motivation for the change was to allow controlling the OpenAPI specification version used, so that `cargo make openapi-install` can always produce the same result as on the current master branch, which could not be guaranteed previously since before the fetch was against the latest commit in the stripe repo.

Specifically, this pr adds the optional argument `--fetch` with the possible values:
- `current`: Use the OpenAPI spec version used to generate the current generated code.
- `latest`: Use the latest OpenAPI release
- `v{}`: Target a specific OpenAPI release.

This could also enable adding future consistency checks to the CI, such as a check that the generated code is up to date with a PR, by comparing generated PR code to `cargo make openapi-install` since spec versions can now be guaranteed to match.

### CODE REVIEW

File: openapi/src/main.rs [hunk 1]
```diff
@@ -1,30 +1,51 @@
...
 mod codegen;
 mod file_generator;
 mod mappings;
 mod metadata;
+mod spec_fetch;
 mod types;
 mod url_finder;
 mod util;
 
+#[derive(Debug, StructOpt)]
```
Comment (mzeitlin11) [inline openapi/src/main.rs:18]: Used `structopt` instead of `clap` because `clap` (and its deps require the 2021 edition)

**Comment by arlyon** [comment]
Hi, sorry for the delay here.

**Comment by arlyon** [comment]
This is an excellent change, I will review shortly.

File: openapi/src/main.rs (hunk 1 above)
Comment (arlyon) [reply openapi/src/main.rs:18]: :+1:

**Comment by arlyon** [comment]
This is great, I had a question around the version file, I take it the goal there is to declaratively specify the version we are using in the repo rather than always fetching latest? I think making it clear in the repo what version we are generating against is the right move but I want to make sure we consider how that impacts the openapi generator action (https://github.com/arlyon/async-stripe/blob/master/.github/workflows/openapi.yml)

**Comment by mzeitlin11** [comment]
> I think making it clear in the repo what version we are generating against is the right move but I want to make sure we consider how that impacts the openapi generator action (https://github.com/arlyon/async-stripe/blob/master/.github/workflows/openapi.yml).

The goal with this change was to maintain behavior for that action (though caveat not sure of a good way to test this :)) There should now be 2 `cargo make` targets with the following behavior:
- `openapi-install`: fetch + generate code for the version in `versions.json`
- `openapi-install-latest`: fetch the latest OpenAPI version and generate code for that. This should also update `versions.json` to that version.

So with these changes the OpenAPI generator action now uses the target `openapi-install-latest` to maintain behavior of trying to generate code for the latest OpenAPI version.

**Comment by arlyon** [comment]
Cool, lets come up with a strategy for updating that version as well.

**Comment by arlyon** [comment]
Merging, ty !:tada:

**Pull request merged**